File: prosecurity.controller/src/kz/diploma/prosecurity/controller/model/UserInfo.java
package kz.diploma.prosecurity.controller.model;

import java.util.HashSet;
import java.util.Set;

public class UserInfo {
  public String displayName;
  public String username;
  public String img;
  public Set<String> cans = new HashSet<>();

  public UserInfo() {}

  public UserInfo(String displayName, String username, String img, Set<String> cans) {
    this.displayName = displayName;
    this.username = username;
    this.img = img;
    this.cans = cans;
  }
}

File: prosecurity.client/src/app/core/model/auth/user-info.ts
export class UserInfo {
  public displayName: string;
  public username: string;
  public img: string;
  public role: string;
  public cans: Set<string>;

  public static of(a: any): UserInfo {
    const ret: UserInfo = new UserInfo();
    ret.assign(a);
    return ret;
  }

  assign(a: any) {
    this.displayName = a.displayName;
    this.username = a.username;
    this.img = a.img;
    this.role = a.role;
    this.cans = (a.cans instanceof Array) ? new Set(a.cans) : new Set();
  }

}
